spec/type_references_and_ownership.md

```diff
@@ -0,0 +1,331 @@
+of `AugmentedKnownTypeReference` we can look up the expected format of the ARM ID for the resource type corresponding to the referenced GVK
+and validate that the customer has provided the correct one.
+
+### A note on the differences between owner references and arbitrary resource references
+Because we are code-generating all of the `ownerRef` fields, we can always supply the annotations for group and kind automatically, using the information
+provided to us by the ARM json specification. **This is not the case for abitrary references (`id`'s) to other resources**. We do not actually know
+programmatically what type that reference is, and it some cases it may actually be allowed to point to multiple different types 
+(think custom image vs shared image gallery).
+
+Currently, the plan is to trust the customer gets it right and ensure we have good
+failure modes for resources if they actually got it wrong (which will cause their resource to move to an unhealthy state). 
+
+## TODOs
+I am planning on adding more details about the below items once I've thought on them more.
+
+- Expand on what happens when a dependent resource is created and its `ownerRef` points to a resource that doesn't exist.
+- Same as above, but for a related resource as opposed to a dependent/owner.
+- Talk about deletion races - if an owning resource is deleted in k8s, Azure deletes (backend) and k8s deletes are all reacing with one another.
+    - What happens if parent (with same name) is recreated immediately in k8s?
+- How are CRD YAMLs going to be rendered into actual ARM deployments - are we making one spec with `dependsOn` or sending many different specs?
+  How are we getting ordering right?
+- How can we allow customers to easily find all dependents for a particular owner (i.e. all subnets of a vnet) using `kubectl`?
+- Test out an `id` detection heuristic.
+- How does this property (which doesn't exist in the Azure ARM spec) get transformed into the right ARM JSON?
+
+## Questions
+These are questions I am posing to the group - I don't expect to have an answer without input from the group.
+
+- Basically everything above is extremely ARM specific. IDs are represented in a different way for data planes and non-ARM endpoints, 
+  and those endpoints JSON schemas may not have the magical `resources` property which allows us to identify owner-dependent relationships.
+  How do we structure the code to solve our ARM problems but allow for alternate solutions for non-ARM?
+- Is subscription a parameter to in YAMLs, or is the operator configured to act only on a certain subscription?
+  If we only operate on a single sub do we not support cross-sub? Do we care?
+- ARM templates have `dependsOn` for specifying creating order of related resources (where A refers to B) - we don't have that, what does that mean?
+- What to do with awkward resources where the owner requires at least 1 dependent to also be created with it?
```

Comment:
One thing I was thinking about is adding an "embedded: true" tag to the `KnownTypeReferences` where the resource must be embedded and having the K8s to ARM converter fetch embedded resources and include them in the deployment template body of the parent.